packages/core/src/amazonq/activation.ts

```diff
@@ -17,6 +17,72 @@ import { listCodeWhispererCommandsWalkthrough } from '../codewhisperer/ui/status
 import { Commands } from '../shared/vscode/commands2'
 import { focusAmazonQPanel, focusAmazonQPanelKeybinding } from '../codewhispererChat/commands/registerCommands'
 import { TryChatCodeLensProvider, tryChatCodeLensCommand } from '../codewhispererChat/editor/codelens'
+import { Auth } from '../auth'
+import { customLearnMoreUri } from '../codewhisperer/models/constants'
+import { openUrl } from '../shared/utilities/vsCodeUtils'
+import { AuthUtil } from '../codewhisperer'
+import { ConnectionStateChangeEvent } from '../auth/auth'
+
+async function showAuthNotification(
+    isAmazonQVisibleEvent: vscode.Event<boolean>,
+    onDidUpdateConnectionEvent: vscode.Event<ConnectionStateChangeEvent | undefined>
+) {
+    let isAmazonQVisible = true // Assume Chat is open by default.
+    let notificationDisplayed = false // Auth Notification should be displayed only once.
+    let authConnection: ConnectionStateChangeEvent
+
+    // Updates the visibility state of the Amazon Q chat.
+    const updateVisibility = async (visible: boolean) => {
+        isAmazonQVisible = visible
+        await showNotification()
+    }
+
+    // Updates the source of the connection for Amazon Q sign in.
+    const updateConnection = async (connection: ConnectionStateChangeEvent | undefined) => {
+        if (connection) {
+            authConnection = connection
+            await showNotification()
+        }
+    }
+
+    const disposables: vscode.Disposable[] = [
+        isAmazonQVisibleEvent(updateVisibility),
+        onDidUpdateConnectionEvent(updateConnection),
+    ]
+
+    async function showNotification() {
+        if (notificationDisplayed || Auth.instance.activeConnection) {
+            return
+        }
+
+        const source = 'authNotification'
+
+        if (!isAmazonQVisible && !authConnection && !AuthUtil.instance.isConnectionExpired()) {
+            const buttonAction = 'Sign In'
+            notificationDisplayed = true
+            disposables.map(item => item.dispose())
```

Comment:
forEach or each() is clearer if this is only done for side effects. same for the other case(s) in this PR
```suggestion
            disposables.each(item => item.dispose())
```